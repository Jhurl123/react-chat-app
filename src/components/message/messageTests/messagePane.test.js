import React from 'react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import MessagePane from '../messagePane'

describe('Message Pane component', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MessagePane />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the ui as expected', () => {
    const tree = renderer
    .create(<MessagePane />)
    .toJSON();
    expect(tree).toMatchSnapshot();  
  })

  it('renders the error message when passed', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MessagePane error={'Couldn\'t send this message!'}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the ui as expected when error is passed', () => {
    const tree = renderer
    .create(<MessagePane error={'Couldn\'t send this message!'} />)
    .toJSON();
    expect(tree).toMatchSnapshot();  
  })
})