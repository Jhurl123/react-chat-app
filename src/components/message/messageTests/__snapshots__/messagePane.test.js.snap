// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Pane component renders the ui as expected 1`] = `
<div
  className="makeStyles-MessagePane-5"
>
  <div>
    <h2
      className="makeStyles-noPrompt-7"
    >
      Click the "Start Conversation" button to connect with new friends!
    </h2>
  </div>
  <ul
    className="makeStyles-container-8"
  />
</div>
`;

exports[`Message Pane component renders the ui as expected when error is passed 1`] = `
<div
  className="makeStyles-MessagePane-5"
>
  <div>
    <div
      className="MuiPaper-root MuiAlert-root MuiAlert-standardError makeStyles-errorMessage-6 MuiPaper-elevation0"
      role="alert"
    >
      <div
        className="MuiAlert-icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      <div
        className="MuiAlert-message"
      >
        <div
          className="MuiTypography-root MuiAlertTitle-root MuiTypography-body1 MuiTypography-gutterBottom"
        >
          Error
        </div>
        Couldn't send this message!
      </div>
    </div>
  </div>
  <div>
    <h2
      className="makeStyles-noPrompt-7"
    >
      Click the "Start Conversation" button to connect with new friends!
    </h2>
  </div>
  <ul
    className="makeStyles-container-8"
  />
</div>
`;
